Plugins/GitUIPluginInterfaces/ObjectId.cs

```diff
@@ -48,322 +49,167 @@ public static ObjectId Random()
-        /// For parsing to succeed, there must be 20 bytes in <paramref name="bytes"/> starting at <paramref name="index"/>.
-        /// </remarks>
-        /// <param name="bytes">The byte array to parse from.</param>
-        /// <param name="index">The index within <paramref name="bytes"/> to commence parsing from.</param>
-        /// <returns>The parsed <see cref="ObjectId"/>.</returns>
-        [MustUseReturnValue]
-        public static ObjectId Parse(byte[] bytes, int index)
-        {
-            return new ObjectId(Read(), Read(), Read(), Read(), Read());
-
-            uint Read() => (uint)((bytes[index++] << 24) |
-                                  (bytes[index++] << 16) |
-                                  (bytes[index++] << 8) |
-                                   bytes[index++]);
-        }
-
-        /// <summary>
-        /// Parses an <see cref="ObjectId"/> from ASCII <paramref name="bytes"/> at the given <paramref name="index"/>.
-        /// </summary>
-        /// <remarks>
-        /// <para>Unlike <see cref="Parse(byte[],int)"/> which reads raw bytes, this method reads human-readable
-        /// ASCII-encoded bytes, which are more verbose. Several git commands emit them in this form.</para>
-        /// <para>For parsing to succeed, there must be 40 bytes in <paramref name="bytes"/> starting at <paramref name="index"/>.</para>
-        /// </remarks>
-        /// <param name="bytes">The byte array to parse from.</param>
-        /// <param name="index">The index within <paramref name="bytes"/> to commence parsing from.</param>
+        /// <param name="array">The char span to parse.</param>
         /// <param name="objectId">The parsed <see cref="ObjectId"/>.</param>
         /// <returns><c>true</c> if parsing succeeded, otherwise <c>false</c>.</returns>
         [MustUseReturnValue]
-        public static bool TryParseAsciiHexBytes(byte[] bytes, int index, [NotNullWhen(returnValue: true)] out ObjectId? objectId)
+        [SuppressMessage("Style", "IDE0057:Use range operator", Justification = "Performance")]
+        public static bool TryParse(in ReadOnlySpan<char> array, [NotNullWhen(returnValue: true)] out ObjectId? objectId)
         {
-            if (index < 0 || index > bytes.Length - Sha1CharCount)
-            {
-                objectId = default;
-                return false;
-            }
-
-            return TryParseAsciiHexBytes(
-                new ArraySegment<byte>(bytes, index, Sha1CharCount),
-                out objectId);
-        }
-
-        [MustUseReturnValue]
-        public static bool TryParseAsciiHexBytes(ArraySegment<byte> bytes, int index, [NotNullWhen(returnValue: true)] out ObjectId? objectId)
-        {
-            // TODO get rid of this overload? slice the array segment instead
-
-            if (bytes.Array is null)
+            if (array.Length != Sha1CharCount)
             {
                 objectId = default;
                 return false;
             }
 
```

Comment:
```suggestion
            if (array.Length != Sha1CharCount
                || ...
```
The positive condtion would be clearer yet.
```suggestion
            if (array.Length == Sha1CharCount
                && ...
            {
                objectId = new ObjectId(i1, i2, i3, i4, i5);
                return true;
            }

```

Reply:
I prefer to have the error exit in the branch, normal action continue.
Two separate error checks.